packages/pixeloven-webpack/file-reporter/src/helper.ts

```diff
@@ -0,0 +1,185 @@
+import filesize from "filesize";
+import fs from "fs";
+import gzipSize from "gzip-size";
+import recursive from "recursive-readdir";
+import { Stats } from "webpack";
+
+import { FileSizeMap, FileSizes, SimplifiedStats } from "./types";
+
+function canReadAsset(asset: string) {
+    return /\.(js|css)$/.test(asset);
+}
+
+function removeFileNameHash(buildFolder: string, fileName: string) {
+    return fileName
+        .replace(buildFolder, "")
+        .replace(/\\/g, "/")
+        .replace(
+            /\/?(.*)(\.[0-9a-f]+)(\.chunk)?(\.js|\.css)/,
+            (match, p1, p2, p3, p4) => p1 + p4,
+        );
+}
+
+export function getDifferenceLabel(currentSize: number, previousSize: number) {
+    const difference = currentSize - previousSize;
+    const fileSize = !Number.isNaN(difference) ? filesize(difference) : 0;
+    if (difference > 0) {
+        return (
+            chalk.dim("(") + chalk.redBright(`+${fileSize}`) + chalk.dim(")")
+        );
+    } else if (difference < 0) {
+        return chalk.dim("(") + chalk.green(`${fileSize}`) + chalk.dim(")");
+    }
+    return chalk.dim(`(no change)`);
+}
+
+export function measureFileSizes(root: string) {
+    return new Promise<FileSizes>((resolve, reject) => {
+        recursive(root, (err, fileNames) => {
+            if (err) {
+                reject(err);
+            }
+            const sizes: FileSizeMap = new Map();
+            if (!err && fileNames) {
+                const filteredFileNames = fileNames.filter(canReadAsset);
+                for (const fileName of filteredFileNames) {
+                    const contents = fs.readFileSync(fileName);
+                    const key = removeFileNameHash(root, fileName);
+                    sizes.set(key, gzipSize.sync(contents));
+                }
+            }
+            resolve({
+                root,
+                sizes,
+            });
+        });
+    });
+}
+
+const friendlySyntaxErrorLabel = "Syntax error:";
+
+function isLikelyASyntaxError(message: string) {
+    return message.indexOf(friendlySyntaxErrorLabel) !== -1;
+}
+
+/**
+ * @todo clean these rules up and make them more relevant to our env.
+ *      This is a port of https://github.com/facebook/create-react-app/blob/master/packages/react-dev-utils/formatWebpackMessages.js
+ * @param message
+ */
+function formatMessage(message: string) {
+    let lines = message.split("\n");
+
+    // Strip Webpack-added headers off errors/warnings
+    // https://github.com/webpack/webpack/blob/master/lib/ModuleError.js
+    lines = lines.filter(line => !/Module [A-z ]+\(from/.test(line));
+
+    // Transform parsing error into syntax error
+    // TODO: move this to our ESLint formatter?
+    lines = lines.map(line => {
+        const parsingError = /Line (\d+):(?:(\d+):)?\s*Parsing error: (.+)$/.exec(
+            line,
+        );
+        if (!parsingError) {
+            return line;
+        }
+        const [, errorLine, errorColumn, errorMessage] = parsingError;
```

Comment:
Why an empty first index?

Reply:
This is a copy paste from create react app. Likely a param they don't need so they just short handed it.